atc/db/resource_config_test.go

```diff
@@ -93,10 +108,15 @@ var _ = Describe("ResourceConfig", func() {
 					Expect(err).ToNot(HaveOccurred())
 					Expect(createdScope.ResourceID()).To(BeNil())
 					Expect(createdScope.ResourceConfig().ID()).To(Equal(resourceConfig.ID()))
+					seqAfterCreate, err := currentResourceConfigScopesIdSeq()
+					Expect(err).ToNot(HaveOccurred())
 
 					foundScope, err := resourceConfig.FindOrCreateScope(nil)
 					Expect(err).ToNot(HaveOccurred())
 					Expect(foundScope.ID()).To(Equal(createdScope.ID()))
+					seqAfterFind, err := currentResourceConfigScopesIdSeq()
+					Expect(err).ToNot(HaveOccurred())
+					Expect(seqAfterCreate).To(Equal(seqAfterFind))
```

Comment:
I would expect further testing on when you use a different resource config to create scope, its scope id is only increased by 1 since the second time you try to create the scope on line 114, it will get rejected (where the bug was happenning)

Reply:
Added more tests. I incidentally committed new tests with `--amend`, thus not adding a new commit.